specification/containerinstance/resource-manager/Microsoft.ContainerInstance/2017-10-01-preview/containerInstance.json

```diff
@@ -0,0 +1,977 @@
+        },
+        "name": {
+          "type": "string",
+          "description": "The event name."
+        },
+        "message": {
+          "type": "string",
+          "description": "The event message."
+        },
+        "type": {
+          "type": "string",
+          "description": "The event type."
+        }
+      }
+    },
+    "ResourceRequirements": {
+      "description": "The resource requirements.",
+      "type": "object",
+      "required": [
+        "requests"
+      ],
+      "properties": {
+        "requests": {
+          "description": "The resource requests of this container instance.",
+          "$ref": "#/definitions/ResourceRequests"
+        },
+        "limits": {
+          "description": "The resource limits of this container instance.",
+          "$ref": "#/definitions/ResourceLimits"
+        }
+      }
+    },
+    "ResourceRequests": {
+      "description": "The resource requests.",
+      "type": "object",
+      "required": [
+        "memoryInGB",
+        "cpu"
+      ],
+      "properties": {
+        "memoryInGB": {
+          "description": "The memory request in GB of this container instance.",
+          "type": "number",
+          "format": "double"
+        },
+        "cpu": {
+          "description": "The CPU request of this container instance.",
+          "type": "number",
+          "format": "double"
+        }
+      }
+    },
+    "ResourceLimits": {
+      "description": "The resource limits.",
+      "type": "object",
+      "properties": {
+        "memoryInGB": {
+          "description": "The memory limit in GB of this container instance.",
+          "type": "number",
+          "format": "double"
+        },
+        "cpu": {
+          "description": "The CPU limit of this container instance.",
+          "type": "number",
+          "format": "double"
+        }
+      }
+    },
+    "AzureFileVolume": {
+      "description": "The properties of the Azure File volume. Azure File shares are mounted as volumes.",
+      "type": "object",
+      "required": [
+        "shareName",
+        "storageAccountName"
+      ],
+      "properties": {
+        "shareName": {
+          "description": "The name of the Azure File share to be mounted as a volume.",
+          "type": "string"
+        },
+        "readOnly": {
+          "description": "The flag indicating whether the Azure File shared mounted as a volume is read-only.",
+          "type": "boolean"
+        },
+        "storageAccountName": {
+          "description": "The name of the storage account that contains the Azure File share.",
+          "type": "string"
+        },
+        "storageAccountKey": {
+          "description": "The storage account access key used to access the Azure File share.",
+          "type": "string"
+        }
+      }
+    },
+    "EmptyDirVolume": {
+      "description": "The empty directory volume.",
+      "type": "object",
+      "properties": {}
```

Comment:
>"properties": {} [](start = 6, length = 16)

Is there something missing here?

Reply:
Don't think so. But for type object, it gives me warning if I don't have property. How to resolve?

Reply:
This model does not have any properties? Like inheriting properties from another model? If it is really something that could contain anything and it is not strongly typed, you could describe it like this...
``` json
"EmptyDirVolume": {
   "description": "The empty directory volume.",
    "type": "object",
    "additionalProperties": {
        "type": "object"
     },
}
```
... or like this if it is something like a dictionary...
``` json
"EmptyDirVolume": {
   "description": "The empty directory volume.",
    "type": "object",
    "additionalProperties": {
        "type": "string"
     },
}
```